azure/services/privatedns/client.go

```diff
@@ -289,9 +290,9 @@ func (ac *azureClient) CreateOrUpdateRecordSet(ctx context.Context, resourceGrou
 	return resp.RecordSet, nil
 }
 
-func virtualNetworkLinkName(clusterName, resourceGroupName string, isManagementCluster bool) string {
-	if isManagementCluster {
-		return fmt.Sprintf("%s-vnet-link", clusterName)
-	}
-	return fmt.Sprintf("%s-dns-%s-%s", clusterName, resourceGroupName, "vnet-link")
+func virtualNetworkLinkName(resourceGroupName string) string {
+	// The name of the linked VNET is "resourceGroupName-vnet"
+	// This is how CAPZ names the VNET links.
+	// We use the same naming convention to avoid conflicts for private MCs.
+	return fmt.Sprintf("%s-vnet-link", resourceGroupName)
```

Comment:
Is it `resourceGroupName-vnet` or `clusterName-vnet`? In case of management clusters it is usually the same.